### PR TITLE
test: close coverage gaps flagged by the pre-v1 codebase audit

### PR DESCRIPTION
**Pull request by mhenrixon**
## Summary

Three test-coverage gaps flagged by the codebase audit, all in critical subsystems that had weak or absent unit/system-level coverage.

**1. `spec/pgbus/streams/presence_spec.rb`** — NEW, 20 examples. Presence shipped with only an integration spec (PGBUS_DATABASE_URL-gated), so live edits to \`presence.rb\` had zero CI safety net. Covers join/leave/touch/members/count/sweep!, the corrupt-metadata fallback (which is the reason the rescue exists), and the AR-missing configuration error.

**2. \`spec/pgbus/active_job/executor_spec.rb\`** — 8 new examples under \"with uniqueness key in payload\". The Executor had zero Uniqueness coverage despite being the enforcement point for at-most-once job semantics. The most important case is the \`until_executed\` strategy with archive failure: \`job_succeeded\` stays false, the ensure block skips \`release_lock\`, and the next retry still sees the lock. A refactor could break that silently without this test.

**3. \`spec/system/dashboard_edge_cases_spec.rb\`** — NEW, 9 Playwright examples:

- XSS: failed job error message with \`<script>\` tag renders as escaped text on both the dashboard and the failed job show page. Asserts \`window.pwned\` is nil — if ERB escaping were off, the script would execute in the Playwright browser.
- XSS: queue name with angle brackets renders as literal text.
- Pagination: empty failed list, \`page=999\`, \`page=-1\`, \`page=abc\` all render without 500ing.
- Not found: \`/pgbus/jobs/99999\` renders the not-found template.
- Null backtrace: failed job with nil backtrace renders without crashing the \`backtrace.present?\` check.

## Test Counts

| Suite | Before | After |
|---|---|---|
| Unit | 1207 | 1234 |
| System | 92 | 101 |

## Verification

- [x] \`bundle exec rubocop\` passes (285 files, 0 offenses)
- [x] \`bundle exec rspec\` (unit) passes (1234 examples, 0 failures)
- [x] \`bundle exec rspec spec/system/\` passes (101 examples, 0 failures)

## What's next

This is PR 2 of 2 from the audit. PR 1 (#95) covers the correctness + perf fixes (DLQ pagination SQL pushdown, paused_queues Set conversion, archive_batch for discard_all_failed, job_stats queue_name index generator). The remaining audit findings will go into a tracking issue for P1/P2 items and feature gaps (Prometheus metrics endpoint, retry backoff per job class, data_source.rb split, etc).

## Summary by CodeRabbit

* **Tests**
  * Added specs for job execution uniqueness locking covering lock acquire/release across success, retry/exception, lock-fail, and dead-letter paths.
  * Added presence stream tests for join/leave/touch/members/count/sweep, metadata parsing/fallbacks, broadcasting behavior, and DB interaction.
  * Added dashboard system tests for XSS safety, pagination edge cases, and graceful handling of missing backtraces.

### CODE REVIEW

File: spec/pgbus/streams/presence_spec.rb [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+
+      it "logs the failure at debug level with truncation" do
+        fake_logger = double("logger")
+        allow(fake_logger).to receive(:debug).and_yield
+        allow(Pgbus).to receive(:logger).and_return(fake_logger)
+
+        presence.members
+
+        expect(fake_logger).to have_received(:debug)
+      end
```
[the remark this answers is not in the document]
Comment (mhenrixon) [reply spec/pgbus/streams/presence_spec.rb]: Fixed in b343ca8. The test now injects a 500+ char metadata blob, captures the logged message via the debug block, and asserts the first 200 source chars are present while anything past 200 is not. Before, the test only verified `.debug` was called, which would have missed a regression removing the 200-char cap.

File: spec/system/dashboard_edge_cases_spec.rb [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+        @stub_data_source.stats[:failed_count] = 1
+      end
+
+      it "does not 500 on an off-the-end page number" do
+        # page=999 is well past the end; the controller should clamp
+        # or render empty without raising. Status <400 means the view
+        # rendered (Capybara raises on 5xx by default).
+        visit "/pgbus/jobs?page=999"
+        expect(page).to have_css("h1", text: "Jobs")
+      end
```
[the remark this answers is not in the document]
Comment (mhenrixon) [reply spec/system/dashboard_edge_cases_spec.rb:108]: Fixed in b343ca8. Reworded the comment to drop the driver-specific claim and make explicit that the `have_css` H1 assertion is the actual guard — a 500 would prevent the H1 from rendering regardless of driver behavior.
